modules/custom/activity_send/modules/activity_send_email/src/Plugin/QueueWorker/ActivitySendEmailWorker.php

```diff
@@ -262,9 +275,31 @@ private function sendToFrequencyManager(array $parameters) {
 
     // We load all the target accounts.
     if ($target_accounts = $user_storage->loadMultiple($parameters['target_recipients'])) {
+      // Get the group from the activity.
+      $activity = $parameters['activity'];
```

Comment:
Please refactor the code to use `$related_entity = $parameters['activity']->getRelatedEntity()` also, move this $related_entity outside if statement which can be then re-used.

Reply:
Check how it looks now.